SRC/icbacn.F90

```diff
@@ -23,7 +23,15 @@ subroutine cnaupd_c(ido, bmat, n, which, nev, tol, resid, ncv, v, ldv,&
   integer(kind=c_int),          value,              intent(in)    :: lworkl
   real(kind=c_float),           dimension(ncv),     intent(out)   :: rwork
   integer(kind=c_int),                              intent(inout) :: info
-  call cnaupd(ido, bmat, n, which, nev, tol, resid, ncv, v, ldv,&
+  
+  character(len=2):: w
+  integer         :: i
+  
+  do i =1,2
+      w(i:i) = which(i)
+  end do
+
+  call cnaupd(ido, bmat, n, w, nev, tol, resid, ncv, v, ldv,&
               iparam, ipntr, workd, workl, lworkl, rwork, info)
```

Comment:
Copy/convert array argument in local `character(len=2)` : sounds good too if CI is OK !
Could you to that also for parpack (PARPACK/SRC/MPI/icbp*F90) ?

Reply:
Ok! Done